### PR TITLE
test(search): refactor for cypress tests

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

Refactor Cypress tests for the `Search` component to use new cypress-component-react framework for testing.
Add `search.test.js`. 

### Current behaviour

Current process is the regular Cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there is newly added test.file
- [x] Check if the `search.test.js` file passed
- [x] Check all tests were moved/refactored from .feature file -> .test.js file
- [x] Check all unnecessary files were deleted, including .json file
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed

### CODE REVIEW

File: src/components/search/search.test.js [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+        );
+
+        getDataElementByValue("error").trigger("mouseover");
+        tooltipPreview()
+          .should("have.text", testCypress)
+          .should("have.attr", "data-placement", tooltipPositionValue);
+      }
+    );
+
+    // threshold prop isn't working
```
Comment (edleeks87) [inline src/components/search/search.test.js:255]: Was this intentionally left in?

File: src/components/search/search.test.js [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+      searchButton()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onChange callback when a click event is triggered", () => {
```
Comment (edleeks87) [inline src/components/search/search.test.js:331]: click event is triggered?

File: src/components/search/search.test.js [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+
+      searchDefaultInput()
+        .type("1")
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onFocus callback when a click event is triggered", () => {
```
Comment (edleeks87) [inline src/components/search/search.test.js:342]: click event

File: src/components/search/search.test.js [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+
+      searchDefaultInput()
+        .focus()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onBlur callback when a click event is triggered", () => {
```
Comment (edleeks87) [inline src/components/search/search.test.js:353]: click

File: src/components/search/search.test.js [hunk 5]
```diff
@@ -0,0 +1,379 @@
+import * as React from "react";
```
Comment (edleeks87) [inline src/components/search/search.test.js:1]:
```suggestion
import React from "react";
```

File: src/components/search/search.test.js (hunk 1 above)
Comment (DlgSHi) [reply src/components/search/search.test.js:255]: yes it was :)

File: src/components/search/search.test.js (hunk 2 above)
Comment (DlgSHi) [reply src/components/search/search.test.js:331]: fixed

File: src/components/search/search.test.js (hunk 3 above)
Comment (DlgSHi) [reply src/components/search/search.test.js:342]: fixed

File: src/components/search/search.test.js (hunk 4 above)
Comment (DlgSHi) [reply src/components/search/search.test.js:353]: fixed

File: src/components/search/search.test.js (hunk 5 above)
Comment (DlgSHi) [reply src/components/search/search.test.js]: fixed

**Review by edleeks87 — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**